code-challenges.js

```diff
@@ -31,38 +94,167 @@ const temp2 = 350
 
 // Pseudo code:
+// input: array
+// output: The output will return the last index of the given value from the array using the test variables.
+
+// Process: 
+// I will take the declared variable given, and determine the value of the data. 
+// After I have determined I have an array, I see the variable we are wanted to get the index of is in the array twice. 
+// Next I will run the the first declared variable and add the built-in method .lastIndexOf(), Inside the argument I will input the second declared variable to return the last index number.
 
 const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10]
 const givenValue1 = 42
+
+// console.log(myNumbers.indexOf(givenValue1))
+//ouput: 2 // this is because the element 42 is in my array twice, i will have to use the built-in method .lastIndexOf() to get target the index of the second 42
+
+console.log(myNumbers.lastIndexOf(givenValue1))
+//ouput: 7
+
 // Expected output: 7
 
 const givenValue2 = 10
+
+console.log(myNumbers.lastIndexOf(givenValue2))
+//output: 8
+
 // Expected output: 8
 
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
 // Pseudo code:
+// input: array
+// output: The output will sort all the numbers in an array from largest to smallest.
+
+// Process: 
+// I will take the declared variable given, and determine the value of the data. 
+// After I have determined I have an array, I will want to use a built-in method for arrays to sort the numbers in order from smallest to largest. Since .sort() is a mutator the original array will be motified to an array sorted from smallest to largest.
+// Next I will run the test variable given name sanDiegoSummerTemperatures in the first example and sanDiegoWinterTemperatures in the second, and add on the mutator .reverse() to create an output with the values now backwards making them list in order from largest to smallest.
+// To check work we can run our (const) variable name to see the new mutated array.
 
 const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
+// sanDiegoSummerTemperatures.sort((a, b) => a - b)
+
+// var summerTemps = sanDiegoSummerTemperatures.sort()
+// console.log(summerTemps.reverse(sanDiegoSummerTemperatures))
+//output: 
+// [
+//     82, 80, 79, 77,
+//     76, 73, 72
+//   ]
+
+sanDiegoSummerTemperatures.sort()
+console.log(sanDiegoSummerTemperatures.reverse())
+// console.log(sanDiegoSummerTemperatures) // the original array mutated
+//output: 
+// [
+//     82, 80, 79, 77,
+//     76, 73, 72
+//   ]
+
 // Expected output: [82, 80, 79, 77, 76, 73, 72]
 
 const sanDiegoWinterTemperatures = [59, 68, 62, 59, 66, 67, 66]
-// Expected output: [68, 67, 66, 66, 62, 59, 59]
+// sanDiegoSummerTemperatures.sort((a, b) => a - b)
+
+// var winterTemps = sanDiegoWinterTemperatures.sort()
```

Comment:
You are so close here, and I see what you wanted to do! You used the .sort() in the console.log right above it, but not in this function you made. I love this try! It means you do understand!